usr/sbin/p11sak/p11sak.c

```diff
@@ -26,6 +33,54 @@ static const char *default_pkcs11lib = "libopencryptoki.so";
 static void *pkcs11lib = NULL;
 static CK_FUNCTION_LIST *funcs = NULL;
 
+static struct ConfigBaseNode *cfg = NULL;
+
+static void error_hook(int line, int col, const char *msg)
+{
+  fprintf(stderr, "PARSE ERROR: %d:%d: %s\n", line, col, msg);
+}
+
+void dump_attr(CK_ATTRIBUTE_PTR a)
+{
+    const char *typestr;
+    typestr = p11_get_cka(a->type);
+
+    switch (a->ulValueLen) {
+    case 0:
+        printf("          %s: no value\n", typestr);
+        break;
+    case 1:
```

Comment:
You differentiate for the different value length here (1, 2, 4, 8), but still print the values in hex regardless of the length. Wouldn't it be easier to always use the default case code and just print the hex bytes, independent of the value length? You seem to call dump_attr() only for byte-array attribute types, so therefore you could always print the value as hex dump, regardless of the value length. 

The length = 0 case is certainly a special case and its good to print 'no value' in that case.

Reply:
Done